implementations/micrometer-registry-statsd/src/main/java/io/micrometer/statsd/StatsdMeterRegistry.java

```diff
@@ -185,6 +194,17 @@ void poll() {
 
     public void start() {
         if (started.compareAndSet(false, true)) {
+            if (statsdConfig.enabled()) {
+                Sinks.Many<String> sink = Sinks.unsafe().many().unicast().onBackpressureBuffer(new MpscArrayQueue<>(Queues.SMALL_BUFFER_SIZE));
+
+                if (HAS_TURBO_FILTER) {
+                    sink = new LogbackMetricsSuppressingManySink(sink);
+                }
+
+                this.sink = sink;
```

Comment:
Updating `this.sink` here does not update the reference that already registered meters have, which is why some of the `StatsdMeterRegistryPublishTest` tests are failing. For example, when the registry is stopped and then started, `this.sink` will be a new sink, but an already created `StatsdCounter` will still reference the old sink.